org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/Document.xtend

```diff
@@ -41,13 +52,13 @@ import org.eclipse.lsp4j.Range
         if (position.line === line && position.character === column) {
             return l
         }
-        throw new IndexOutOfBoundsException(position.toString + " text was : " + contents)
+        throw new IndexOutOfBoundsException(position.toString + if (printSourceOnError) "" else (" text was : " + contents))
```

Comment:
what is the actual reason you want to change the behaviour.
the index out of bounces should actually never happen

Reply:
It should never happen, but it did in my implementation, probably because of a mistake on my part. However, we have clients where code is never allowed to leak, even when talkback is enabled. A small mistake on our side could mean that some code of one of our clients could leak back to us. I'm convinced that other Xtext users have similar clients and thus want a way to disable this.

Reply:
:+1: